x/validator/handler_test.go

```diff
@@ -177,3 +186,360 @@ func queryValidator(setup TestSetup, owner string) (*types.Validator, error) {
+}
+
+// func TestHandler_NodeAdminCantEnableValidatorDisabledByTrustees(t *testing.T) {
+// 	setup := Setup(t)
+
+// 	// create Trustee and NodeAdmin
+// 	ba1 := authtypes.NewBaseAccount(testconstants.Address1, testconstants.PubKey1, 0, 0)
+// 	account1 := dclauthtypes.NewAccount(ba1,
+// 		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID1)
+// 	setup.DclauthKeeper.SetAccount(setup.Ctx, account1)
+
+// 	ba2 := authtypes.NewBaseAccount(sdk.AccAddress(testconstants.ValidatorAddress1), testconstants.PubKey2, 0, 0)
+// 	account2 := dclauthtypes.NewAccount(ba2,
+// 		dclauthtypes.AccountRoles{dclauthtypes.NodeAdmin}, nil, testconstants.VendorID2)
+// 	setup.DclauthKeeper.SetAccount(setup.Ctx, account2)
+
+// 	// propose and approve new disablevalidator (will be approved because of 1 trustee)
+// 	msgProposeDisableValidator := NewMsgProposeDisableValidator(account1.GetAddress(), sdk.ValAddress(testconstants.ValidatorAddress1))
+// 	_, err := setup.Handler(setup.Ctx, msgProposeDisableValidator)
+// 	require.NoError(t, err)
+
+// 	msgEnableValidator := types.NewMsgEnableValidator(sdk.ValAddress(account2.GetAddress()))
+// 	_, err = setup.Handler(setup.Ctx, msgEnableValidator)
+// 	require.Error(t, err)
+// 	require.ErrorIs(t, err, sdkerrors.ErrUnauthorized)
+// }
+
+func TestHandler_DisabledValidatorAlreadyExistsPropose(t *testing.T) {
+	setup := Setup(t)
+
+	// create Trustees
+	ba1 := authtypes.NewBaseAccount(testconstants.Address3, testconstants.PubKey3, 0, 0)
+	account1 := dclauthtypes.NewAccount(ba1,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID3)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account1)
+
+	ba2 := authtypes.NewBaseAccount(testconstants.Address2, testconstants.PubKey2, 0, 0)
+	account2 := dclauthtypes.NewAccount(ba2,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID2)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account2)
+
+	// propose and approve new disablevalidator (will be approved because of 2 trustees)
+	msgProposeDisableValidator := NewMsgProposeDisableValidator(account1.GetAddress(), sdk.ValAddress(testconstants.ValidatorAddress1))
+	_, err := setup.Handler(setup.Ctx, msgProposeDisableValidator)
+	require.NoError(t, err)
+
+	msgProposeDisableValidator = NewMsgProposeDisableValidator(account1.GetAddress(), sdk.ValAddress(testconstants.ValidatorAddress1))
+	_, err = setup.Handler(setup.Ctx, msgProposeDisableValidator)
+	require.Error(t, err)
+	require.ErrorIs(t, err, types.ErrDisabledValidatorAlreadytExists)
+}
+
+func TestHandler_DisabledValidatorAlreadyExistsDisable(t *testing.T) {
+	setup := Setup(t)
+
+	ba3 := authtypes.NewBaseAccount(sdk.AccAddress(testconstants.ValidatorAddress1), testconstants.PubKey3, 0, 0)
+	account3 := dclauthtypes.NewAccount(ba3,
+		dclauthtypes.AccountRoles{dclauthtypes.NodeAdmin}, nil, testconstants.VendorID3)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account3)
```

Comment:
The node admin account should be created in `Setup` function together with the validator he/she owns (see the corresponding comment in `Setup` function).